fairlearn/datasets/_fetch_data.py

```diff
@@ -0,0 +1,91 @@
+import warnings
+from ._common import FairnessWarning
+
+from sklearn.datasets import fetch_openml
+
+
+def fetch_boston(*, version='1', cache=True, data_home=None,
+                 as_frame=False, return_X_y=False, warn=True):
+    """Load the boston housing dataset (regression).
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total               506
+    Dimensionality               13
+    Features                   real
+    Target           real 5. - 50.
+    ==============   ==============
+
+    This dataset has known fairness issues. There's a "lower status of
+    population" (LSTAT) parameter that you need to look out for and a column
+    that is a derived from the proportion of people with a black skin color
+    that live in a neighborhood (B). This dataset should be used to explore
+    and benchmark fairness. Predicting a house-price while ignoring these
+    sensitive attributes is naive.
```

Comment:
This paragraph needs a bit of tweaking. Mostly, the user needs to know where to go for details about the data description and the fairness issues.

Two clarification questions:
* By explore/benchmark fairness do you mean "explore/benchmark unfairness mitigation algorithms"?
* In the last sentence, by "naive" do you mean the fact that many other features can serve as proxies for sensitive features, so the strategy of simply removing sensitive features is often an ineffective mitigation strategy?

Reply:
I've though about it and came to the conclusion that the issues with the dataset deserve an article. For now I've changed the information here to include references that folks can read. 

Benchmarking fairness deserves a seperate page on the docs as well, explaining that is something that should not go here.